### PR TITLE
[REF] minor cleanup of groupBy definition.

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Minor cleanup of variables - this is preliminary to extracting the group function

Before
----------------------------------------
variables not logically grouped 

After
----------------------------------------
variables more logically grouped

Technical Details
----------------------------------------
If you read the code the groupBy statement creation can go with 'its friends'
and the groupBy in the removed line is overwritten. I'll extract this next ....

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Contact/BAO/Query.php [hunk 1]
```diff
@@ -4889,7 +4889,6 @@ public function searchQuery(
     $groupBy = $groupByCols = NULL;
     if (!$count) {
       if (isset($this->_groupByComponentClause)) {
-        $groupBy = $this->_groupByComponentClause;
```
Comment (eileenmcnaughton) [inline CRM/Contact/BAO/Query.php]: this is then overwritten using the groupByCols values so we don't need it

File: CRM/Contact/BAO/Query.php [hunk 2]
```diff
@@ -4916,7 +4918,7 @@ public function searchQuery(
     //      MySQL expect the columns present in GROUP BY, must be present in SELECT clause and that results into error, needless to have other columns.
     //   2. When GROUP BY columns are present then disable FGB otherwise it demands to add ORDER BY columns in GROUP BY and eventually in SELECT
     //     clause. This will impact the search query output.
-    $disableFullGroupByMode = ($sortByChar || !empty($groupByCols) || $groupContacts);
+    $disableFullGroupByMode = ($sortByChar || !empty($groupBy) || $groupContacts);
```
Comment (eileenmcnaughton) [inline CRM/Contact/BAO/Query.php]: this has the same effect - I want to extract the above so less functions = better

File: CRM/Contact/BAO/Query.php [hunk 3]
```diff
@@ -4930,10 +4932,6 @@ public function searchQuery(
 
     list($select, $from, $where, $having) = $this->query($count, $sortByChar, $groupContacts, $onlyDeleted);
 
-    if (!empty($groupByCols)) {
-      $groupBy = " GROUP BY " . implode(', ', $groupByCols);
```
Comment (eileenmcnaughton) [inline CRM/Contact/BAO/Query.php]: moved higher up - not touched in any meaningful way in-between

**Comment by colemanw** [comment]
Merging based on code review & passing tests.

**Pull request merged**